CONTRIBUTING.md

```diff
@@ -157,6 +157,18 @@ If problems are found, please report these problems to the grammar maintainer as
 
 Please then open a pull request for the updated grammar.
 
+## Changing the color of a language
+
+Many of the colors associated with the languages within Linguist have been in place for a very long time.
+The colors were often chosen based on the colors used by the language at the time and since then users will have become familiar with those colors as they appear on GitHub.com.
+If you would like to change the color of a language, we ask that you propose your suggested color change to the wider community for your language to gain consensus before submitting a pull request.
+Please do this in a community forum or repository used and known by the wider community of that language, not the Linguist repository.
+
+Once you've received consensus that the community is happy with your proposed color change, please feel free to open a PR making the change and link to the public discussion where this was agreed by the community.
+If there are official branding guidelines to support the colour choice, please link to those too.
+
+Please note that Linguist currently implements a [color proximity test][color-proximity-test] to ensure colors are sufficiently different from one another so you may not be able to use the precise color you want - reds and blues are really popular.
+As such, we recommend you test the color change locally before making your plea to the wider language community.
```

Comment:
Before we promulgate this as a restriction, we should *really* wait for the final consensus as to whether or not the colour-proximity requirements will ever be axed in future.

We have no idea how long it might take to reach a consensus. Members of a language's community could spend weeks bikeshedding as to whether a compromise for their project's official branding should be taken, simply their shade of red was too close to 5-6 other shades of red. How bad would we (GitHub/Linguist) look if the colour restriction was suddenly abolished the day after they reached an agreement?

Obviously we can't expect it to take *that* long to decide on a hex colour value, but you know what people are like...

Reply:
> Before we promulgate this as a restriction, we should _really_ wait for the final consensus as to whether or not the colour-proximity requirements will ever be axed in future.

We may be waiting a while. I've given my issue a nudge to see if I can get a quick response, but it's in a long list of other higher priority issues. The best thing to do is assume the status quo for now, we can easily remove the note when we remove the test.

Reply:
Fair enough, then.

I might be getting anxious for the outcome because there's a lot hanging on the removal of proximity requirements. We can *finally* degroup the languages touched on in #4291, add official colours for data/prose formats (which matters now that we have `linguist-detectable`), and a whole heap of other nags...